tests/end_to_end/helpers.h

```diff
@@ -1171,10 +953,6 @@ typedef class _program_info_provider
 
         if (custom_program_data != nullptr) {
             program_data = custom_program_data;
-        } else if (program_type == EBPF_PROGRAM_TYPE_XDP) {
-            program_data = &_mock_xdp_program_data;
-        } else if (program_type == EBPF_PROGRAM_TYPE_XDP_TEST) {
-            program_data = &_ebpf_xdp_test_program_data;
```

Comment:
question: if we are removing this mock provider, how are the libbpf unit tests for xdp specific APIs working?